tests/test_split_command.rs

```diff
@@ -132,3 +132,131 @@ fn test_split_by_paths() {
     A file2
     "###);
 }
+
+#[test]
+fn test_split_empty_child() {
+    let test_env = TestEnvironment::default();
+    test_env.jj_cmd_success(test_env.env_root(), &["init", "repo", "--git"]);
+    let repo_path = test_env.env_root().join("repo");
+
+    std::fs::write(repo_path.join("file1"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "original parent"]);
+
+    std::fs::write(repo_path.join("file2"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["commit", "-m", "target commit"]);
+
+    std::fs::write(repo_path.join("file3"), "foo").unwrap();
+    test_env.jj_cmd_success(&repo_path, &["describe", "-m", "first original child"]);
+
+    test_env.jj_cmd_success(
+        &repo_path,
+        &["new", "-r", "@-", "-m", "second original child"],
+    );
+    std::fs::write(repo_path.join("file3"), "foo").unwrap();
+
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "log",
+            "-T",
+            r#"change_id.short() if(empty, " (empty) ", " ") description"#,
+        ],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    @ d043564ef936 second original child
+    | o ffdaa62087a2 first original child
+    |/  
+    o 8e4fac809cbb target commit
+    o 9a45c67d3e96 original parent
+    o 000000000000 (empty) (no description set)
+    "###);
+
+    let stdout = test_env.jj_cmd_success(&repo_path, &["split", "-r", "@-", "--empty-child"]);
+    insta::assert_snapshot!(stdout, @r###"
+    Rebased 2 descendant commits
+    First part: 198584685467 target commit
+    Second part: 1c004c965469 (no description set)
+    Working copy now at: a3941c48999c second original child
+    "###);
+    let stdout = test_env.jj_cmd_success(
+        &repo_path,
+        &[
+            "log",
+            "-T",
+            r#"change_id.short() if(empty, " (empty) ", " ") description"#,
+        ],
+    );
+    insta::assert_snapshot!(stdout, @r###"
+    @ d043564ef936 second original child
+    | o ffdaa62087a2 first original child
+    |/  
+    o 19b790168e73 (empty) (no description set)
+    o 8e4fac809cbb target commit
+    o 9a45c67d3e96 original parent
+    o 000000000000 (empty) (no description set)
+    "###);
+}
```

Comment:
Optional: You could do a `jj undo` here and then include the second test, which would be shorter. I also keep finding weird bugs with `jj undo` by structuring tests this way. I find this doesn't make tests less readable. Because of `insta`, this usually doesn't cause the entire test to fail when a piece of it fails (though sometimes it does happen). If the test gets long, I just insert another `jj log` after the `jj undo`.

Reply:
I have added a helper to create the initial setup (two parents, target commit, two children) so the code is lighter. But I'll keep the `undo` in mind.